app/controllers/collections_controller.rb

```diff
@@ -170,4 +171,25 @@ def dataset_download
     end
     send_file zip_file
   end
+
+  def build_citation collection
+    return nil unless (!collection.identifier.blank? && collection.identifier.first.include?("doi:"))
+    datacite_api_base = "https://api.datacite.org/application/vnd.schemaorg.ld+json/"
+    datacite_id = collection.identifier.first.gsub("doi:", "")
+    datacite_api_url = File.join(datacite_api_base, datacite_id)    
+    begin
+      response = HTTParty.get(datacite_api_url)   
+      datacite_record = JSON.parse(response.body)
+    rescue
+      logger.error "Error fetching datacite record for collection"
+    end
+    return nil unless  !datacite_record.nil?
+    creator = "#{datacite_record['author']['familyName']}, #{datacite_record['author']['givenName'][0]}."
```

Comment:
You might want to check the validity of the datacite doi, for example, if I used an invalid doi such as "doi:10.7294/w4bg2kws", the application will throw an error with the datacite_record not being nil.